### PR TITLE
#108: [Dropdown] defects (closes #108)

### LINKED ISSUE #108: [Dropdown] defects
## Requirements
- When text input is dirty and gets in `blur` state, the `Dropdown` keeps displaying the value in the input. It should instead display the current selected option (or the placeholder 'Select...')
- When in focus, selected option or placeholder should have reduced opacity (placeholderish) to induce the user to understand he can type text to search options

### PR DESCRIPTION
**Pull request by nerder**
Issue #108

### CODE REVIEW

File: examples/components/Dropdown/default.example [hunk 1]
```diff
@@ -7,6 +12,7 @@ class Example extends React.Component {
       { value: 'orange', label: 'Orange' },
       { value: 'banana', label: 'Banana' }
     ];
+    //const valueLink = { value: this.state.value, requestChange: value => this.setState({ value }) };
```
Comment (FrancescoCioria) [inline examples/components/Dropdown/default.example]: remove?

File: src/dropdown/Dropdown.js [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
 import { warn } from '../utils/log';
 
-const themes = {
-  semantic: 'semantic-theme'
-};
-
 const PropTypes = {
-  children: t.maybe(t.ReactNode),
-  theme: t.maybe(t.enums.of(['semantic'])),
-  valueLink: t.maybe(t.struct({
```
Comment (FrancescoCioria) [inline src/dropdown/Dropdown.js]: why did you remove `valueLink`? please revert this

File: src/dropdown/Dropdown.js [hunk 3]
```diff
@@ -24,34 +22,46 @@ export default class Dropdown extends React.Component {
...
-  render() {
-    // The order is important: props may override previous ones
-    return (
-      <Select
-        {...this.getGeneralProps()}
-        {...this.getValueLinkProps()}
-        {...this.getClassName()}
-      />
-    );
+  _onChange = () => this.props.valueLink ? this.props.valueLink.requestChange : this.props.onChange;
```
Comment (FrancescoCioria) [inline src/dropdown/Dropdown.js]: 1. `_onChange` is called with a `value`
2. this is an usual why of defining a `_onChange` function: why aren't you calling the prop `onChange` here?
I think you should do this:
```js
_onChange(value) => this.props.valueLink ? this.props.valueLink.requestChange(value) : this.props.onChange(value);
```

or if you prefer your version then please rename it to `getOnChange` (as that's what you're actually doing)
this.props.valueLink ? this.props.valueLink.requestChange() : this.props.onChange;

File: src/dropdown/Dropdown.js [hunk 4]
```diff
@@ -24,34 +22,46 @@ export default class Dropdown extends React.Component {
...
+      valueLink,
+      ...props
+    } = this.props;
+
+    return {
+      ...props,
+      options,
+      className: cx('dropdown', className),
+      value: this.valueToOption(this.getValue(), options),
+      onChange: this._onChange(),
```
Comment (FrancescoCioria) [inline src/dropdown/Dropdown.js]: as pointed in the comment above this should be either `this._onChange` or `this.getOnChange()`

File: src/dropdown/Dropdown.js (hunk 3 above)
Comment (nerder) [inline src/dropdown/Dropdown.js]: actually i prefer my version that return the function.
I should rename it then

File: src/dropdown/Dropdown.js [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
-const themes = {
-  semantic: 'semantic-theme'
-};
-
 const PropTypes = {
-  children: t.maybe(t.ReactNode),
-  theme: t.maybe(t.enums.of(['semantic'])),
+  value: t.maybe(t.union([t.Number, t.String, t.Object])),
   valueLink: t.maybe(t.struct({
     value: t.maybe(t.String),
```
Comment (FrancescoCioria) [inline src/dropdown/Dropdown.js]: as you improved the evaluation for `value`, can you improve this too?
(`value: t.maybe(t.union([t.Number, t.String, t.Object])),`)

**Comment by FrancescoCioria** [comment]
:kimono:
